csharp/src/Apache.Arrow.Flight/Apache.Arrow.Flight.csproj

```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+    <LangVersion>8.0</LangVersion>
```

Comment:
If it just works - moving this change to the Directory.Build.props would be fine with me. That way we are using the latest C# for all projects.

Reply:
This was changed in #8702  so it can be tested/approved seperately.